__init__.py

```diff
@@ -183,12 +195,28 @@ def weather_at_location(self, name):
                 return self.weather_at_location(name)
             raise
 
+    def check_for_city_state(self, name):
+        # someone might ask for denver colorado (which would need to be written denver,colorado)
+        reg = self.owm.city_id_registry()
+        if len(reg.ids_for(name)) == 0:
+            # Try to rewrite name -> replace last blank with comma # TODO: careful this might be language specific -> check
+            p = name.rfind(" ")
+            if p>0:
+                name = name[:p] + "," + name[p+1:]
```

Comment:
Is it worth rechecking whether this new name variation exists in the registry? 
If not we can perhaps return a location not found error earlier.

Reply:
Yep, I am not sure how your location registry actually works and what is in there - so maybe there would be a better solution than this.

Reply:
I don't know much about the registry apart from what is in the [pyowm docs](https://pyowm.readthedocs.io/en/latest/usage-examples-v2/weather-api-usage-examples.html?highlight=city_id_registry#retrieving-city-id-for-a-location)

This is a broader problem which is why we added a geolocation endpoint to Mycroft. It's not an easy thing to parse a string and return an exact location. There's always multiple places with the same name, an unknown number of words in each part of the location, and you don't know what combination of parts people are saying eg suburb + country, city + state,  city + state + country etc.

Happy to include this in the interim as it does check another variation of possibilities, but ultimately I think we need to move to a complete location parser. Translating the location to coordinates means we can be more confident in the response from OWM.